src/core/quantum_gate.jl

```diff
@@ -1146,6 +1163,29 @@ Base.inv(gate::ISwap) = iswap_dagger(gate.target_1,gate.target_2)
 
 get_connected_qubits(gate::ISwap)=[gate.target_1, gate.target_2]
 
+"""
+    swap(qubit_1, qubit_2)
+
+Return the swap `Gate` which applies the swap `Operator` to `qubit_1` and `qubit_2.`
+
+The corresponding `Operator` is [`swap()`](@ref).
+""" 
+function swap(qubit_1, qubit_2)
+    ensure_target_qubits_are_different([qubit_1, qubit_2])
+    return Swap(qubit_1, qubit_2)
+end
+
+struct Swap <: AbstractGate
+    target_1::Int
+    target_2::Int
+end
+
+get_operator(::Swap, T::Type{<:Complex}=ComplexF64) = swap(T)
+
+Base.inv(gate::Swap) = swap(gate.target_1,gate.target_2)
+
+get_connected_qubits(gate::Swap)=[gate.target_1, gate.target_2]
```

Comment:
add an entry in @testset 'inv' in test/gates.jl for swap gate, calling Base.inv() and get_connected_qubits()